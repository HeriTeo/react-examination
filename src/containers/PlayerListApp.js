import React, { Component } from 'react';
import { connect } from 'react-redux';
import styles from './PlayerListApp.css';
import {
  addPlayer, deletePlayer, starPlayer, paginatePlayer, clickPage,
} from '../actions/PlayersActions';
import {
  PlayerList, AddPlayerInput, Pagination, Selection,
} from '../components';

class PlayerListApp extends Component {
  constructor() {
    super();
    this.state = {
      selected: 'ALL',
    }
  }
  componentDidMount() {
    const { playerlist: { playersById }, paginatePlayer } = this.props;
    paginatePlayer(playersById);
  }

  onSelectData(selected) {
    const { playerlist: { playersById }, paginatePlayer, clickPage } = this.props;
    let filterPlayer = playersById;
    if (selected !== "ALL") {
      filterPlayer = playersById.filter(player => selected.indexOf(player.position) !== -1);
    }
    
    clickPage(0);
    paginatePlayer(filterPlayer);
    this.setState({ selected });
  }

  render() {
    const {
      playerlist: { playersById, paginationPage, currentPage },
    } = this.props;

    const {
      selected,
    } = this.state;

    const actions = {
      addPlayer: this.props.addPlayer,
      deletePlayer: this.props.deletePlayer,
      starPlayer: this.props.starPlayer,
      paginatePlayer: this.props.paginatePlayer,
      clickPage: this.props.clickPage,
    };

    return (
      <div className={styles.playerListApp}>
        <h1>NBA Players</h1>
        <AddPlayerInput
          addPlayer={actions.addPlayer}
          paginatePlayer={actions.paginatePlayer}
          players={playersById}
        />
        <Selection 
          onSelectData={data => this.onSelectData(data)}
        />
        <PlayerList
          selected={selected}
          currentPage={currentPage}
          paginationPage={paginationPage}
          players={playersById}
          actions={actions}
        />
        <Pagination 
          clickPage={actions.clickPage}
          paginationPage={paginationPage} 
        />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return state;
}

export default connect(
  mapStateToProps,
  {
    addPlayer,
    deletePlayer,
    starPlayer,
    paginatePlayer,
    clickPage,
  },
)(PlayerListApp);
