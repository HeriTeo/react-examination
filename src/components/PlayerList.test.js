import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import PlayerListApp from './PlayerList';

configure({ adapter: new Adapter() });

describe('PlayerList', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<PlayerListApp />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
