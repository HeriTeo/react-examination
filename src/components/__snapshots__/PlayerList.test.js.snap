// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerList renders without crashing 1`] = `<ul />`;
